src/settings.c

```diff
@@ -100,52 +145,50 @@ void check_and_correct_settings(struct settings *s) {
         }
 }
 
-void load_settings(char *cmdline_config_path)
-{
-
-        FILE *config_file = NULL;
-
-        if (cmdline_config_path) {
-                if (STR_EQ(cmdline_config_path, "-")) {
-                        config_file = stdin;
-                } else {
-                        config_file = fopen(cmdline_config_path, "r");
-                }
+static void apply_settings(FILE *config_file) {
+        if (config_file)
+                load_ini_file(config_file);
 
-                if (!config_file) {
-                        DIE("Cannot find config file: '%s'", cmdline_config_path);
-                }
-        }
-
-        if (!config_file) {
-                config_file = xdg_config("/dunst/dunstrc");
-        }
-
-        if (!config_file) {
-                /* Fall back to just "dunstrc", which was used before 2013-06-23
-                 * (before v0.2). */
-                config_file = xdg_config("/dunstrc");
-        }
-
-        if (!config_file) {
-                LOG_W("No dunstrc found.");
-        }
-
-        load_ini_file(config_file);
         settings_init();
         set_defaults();
         save_settings();
-
         check_and_correct_settings(&settings);
 
         for (GSList *iter = rules; iter; iter = iter->next) {
                 struct rule *r = iter->data;
                 print_rule(r);
         }
 
-        if (config_file) {
+        if (config_file)
                 fclose(config_file);
-                free_ini();
+
+        free_ini();
+}
+
+void load_settings(char *cmdline_config_path) {
+        FILE *config_file = NULL;
+        if (cmdline_config_path) {
+                if (STR_EQ(cmdline_config_path, "-"))
+                        config_file = stdin;
+                else
+                        config_file = fopen(cmdline_config_path, "r");
+
+                if (config_file) {
+                        apply_settings(config_file);
```

Comment:
The config given by the command line should be the most important config file. Now it's the least important

Reply:
No, it should be the *only* one in case it can be opened, because then load_settings() returns early.

Reply:
Oh right